src/core/CorePool.ts

```diff
@@ -325,17 +419,68 @@ export class CorePool {
     tickUpper: number,
     liquidityDelta: JSBI
   ): Position {
-    // TODO
-    return new Position();
+    let position: Position = this.getPosition(owner, tickLower, tickUpper);
```

Comment:
this.positionManager.getPositionAndInitIfAbsent(owner, tickLower, tickUpper);
Because `this.getPosition` is for view use and won't add new `Position` to `PositionManager`.

Reply:
Ok, within updatePositioin we indeed need a new obj of `Position`